src/docker/restorableappengine.cc

```diff
@@ -756,6 +786,12 @@ void RestorableAppEngine::verifyComposeApp(const std::string& compose_cmd, const
        boost::process::start_dir = app_dir);
 }
 
+void RestorableAppEngine::pullComposeAppImages(const std::string& compose_cmd, const boost::filesystem::path& app_dir,
+                                               const std::string& flags) {
+  exec(boost::format{"%s pull %s"} % compose_cmd % flags, "failed to pull Compose App images",
+       boost::process::start_dir = app_dir);
```

Comment:
Just to make sure I'm following this: The "pullImage" method called by `pullAppImages` called by `fetch` will have loaded everything. This method is really just updating docker with tag information which requires a network hop?

Maybe a comment or something in code would help or calling the method something different?

Reply:
>The "pullImage" method called by pullAppImages called by fetch will have loaded everything. This method is really just updating docker with tag information which requires a network hop?

Correct (hash/digest not tag to be precise).

Reply:
>>Maybe a comment or something in code would help or calling the method something different?

I'll add comment.

Reply:
Also, the `install` and `run` method implementation is almost the same, I think I'll add common method for them in a separate commit.

Reply:
@doanac updated by a force push.